hi-ml-histopathology/src/histopathology/configs/classification/custom_subset/dump_dataset_csv.py

```diff
@@ -0,0 +1,43 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+import argparse
+import os
+import pandas as pd
+import json
+import logging
+from health_azure.utils import is_local_rank_zero
+
+from health_ml.utils.common_utils import logging_to_stdout
+
+
+def dump_dataset_csv_from_wsi_ids(
+    datalist_json: str, dataset_csv: str, dest_dataset_csv: str, subset: str, image_id: str = "slide_id"
+) -> None:
+
+    # get target wsi ids
+    with open(datalist_json, "r") as f:
+        slides = json.load(f)
+    wsi_ids = [slide["image"].split(".")[0] for slide in slides[subset]]
+
+    # select corresponding tiles from dataset
+    df = pd.read_csv(dataset_csv)
+    df.to_csv(dataset_csv, index=False)
+    df.loc[df[image_id].isin(wsi_ids)].to_csv(dest_dataset_csv, index=False)
+
+
+if __name__ == "__main__":
```

Comment:
Can this subsetting script actually be run on-the-fly? At present, you need a slides list, run this script to create a tiles list. But the operations are so cheap that we could do that subsetting in the dataloader.